File: GEMMI/parse_mmcif.cc
// Benchmark the parsing of a mmCIF file given as an argument

#include <gemmi/mmread.hpp>
#include <iostream>
#include <time.h>

int main( int argc, char* argv[] ) {
    std::string mmcif_filepath = argv[1];
    struct timespec tstart, tend;
    clock_gettime(CLOCK_REALTIME, &tstart);
    gemmi::Structure st = gemmi::read_structure_file(mmcif_filepath);
    clock_gettime(CLOCK_REALTIME, &tend);
    printf("%.6f\n", (tend.tv_sec - tstart.tv_sec) + (tend.tv_nsec - tstart.tv_nsec)/1E9);
}

File: biotite/parse_mmcif.py
# Benchmark the parsing of a mmCIF file given as an argument

import sys
import time
import biotite.structure.io.pdbx as pdbx

mmcif_filepath = sys.argv[1]

start = time.time()
file = pdbx.PDBxFile()
file.read(mmcif_filepath)
pdbx.get_structure(file)
end = time.time()

print(end - start)
